File: 3959-check-good-integer/3959-check-good-integer.java
class Solution {
    public boolean checkGoodInteger(int n) {
        int d_sum=0;
        int SQ_sum=0;
        while(n>0){
            int d=n%10;
            d_sum+=d;
            SQ_sum+=(d*d);
            n=n/10;
        }
        if(SQ_sum-d_sum>=50){
            return true;
        }
        else{
            return false;
        }
    }
}

File: 3959-check-good-integer/3959-check-good-integer.py
class Solution:
    def checkGoodInteger(self, n: int) -> bool:
        res=False
        sum=0
        S_sum=0
        while n>0:
            d=n%10
            sum+=d
            S_sum+=(d**2)
            n//=10
        if S_sum-sum>=50:
            res=True
        return res